model/llm/llm.go

```diff
@@ -182,6 +219,59 @@ func (m *Model) queryModel(log *log.Logger, request string) (assessment metrics.
 	return assessment, testContent, nil
 }
 
+// repairSourceCodeFile asks the model to repair the code in case any compilation error occurs in a repository source code.
+func (m *Model) repairSourceCodeFile(ctx task.Context) (assessment metrics.Assessments, err error) {
+	if !strings.HasSuffix(ctx.FilePath, ".go") && !strings.HasSuffix(ctx.FilePath, ".java") {
+		return nil, nil
+	}
+
+	if strings.HasSuffix(ctx.FilePath, "_test.go") || strings.HasSuffix(ctx.FilePath, "Test.java") {
+		return nil, nil
+	}
```

Comment:
This task is called for each file in a repository. Until know, all the repositories only had source files (`.go` or `.java`) but now we also have tests inside a repository (`_test.go` or `Test.java`) so we need to skip running the task for those files. 

Wondering if there is a more elegant way to do this.